src/pages/Jobs.js

```diff
@@ -2,27 +2,52 @@ import React, { useState, useEffect } from "react";
 import JobList from "../components/JobCard";
 import { faSearch } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import { Row, Col, Container } from "react-bootstrap";
+import { Row, Col, Container, Form } from "react-bootstrap";
+import { useHistory } from "react-router-dom";
+
 export default function Jobs() {
   let originalList = [];
-  let filteredList = [];
   let [jobs, setJobs] = useState([]);
-  let keyword = "";
+  let [keyword, setKeyword] = useState("");
+  const [encode, setEncode] = useState("");
```

Comment:
why do you need to keep keyword and encode separately?

Reply:
yeah i remove the first keyword

Reply:
yeah you are right i don't need encode so i remove it

Reply:
actually we need it because encode is use for  call the useEffect. useEffect will be called depends on the encode changes.